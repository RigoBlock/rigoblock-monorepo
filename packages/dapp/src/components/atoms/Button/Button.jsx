import MaterializeButton from '@material-ui/core/Button'
import PropTypes from 'prop-types'
import React from 'react'
import classNames from 'classnames'

export const BUTTON_SIZES = {
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large'
}

const Button = ({ size, text }) => {
  const classProps = classNames('btn')
  return (
    <MaterializeButton variant="raised" classes={classProps}>
      {text}
    </MaterializeButton>
  )
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
  size: PropTypes.oneOf([
    BUTTON_SIZES.SMALL,
    BUTTON_SIZES.MEDIUM,
    BUTTON_SIZES.LARGE
  ])
}

Button.defaultProps = {
  size: BUTTON_SIZES.MEDIUM
}

export default Button
