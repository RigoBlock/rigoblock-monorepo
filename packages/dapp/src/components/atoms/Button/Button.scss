@import "../../_settings/variables.scss";
@import '~react-md/src/scss/react-md';
@import '~react-md/src/scss/globals';
@import '~react-md/src/scss/colors';

@include react-md-buttons($md-primary-color, $md-secondary-color);
@include react-md-accessibility;
@include react-md-transitions;
@include react-md-grid;
@include react-md-inks;
@include react-md-menus;
@include react-md-lists;
@include react-md-dividers;
@include react-md-icons;
@include react-md-layovers;
@include react-md-select-fields;
@include react-md-helpers-everything;
@include react-md-text-fields;
@include react-md-colors;

.base-button {
  border: solid $border-size;
  font-weight: $font-bold;
  padding: ($dist-xs - $border-size) ($dist-s - $border-size)
}

.inverted-button {
  border: none;
  font-weight: $font-bold;
}

.success-button {
  background: $success-color;
  &:hover {
    background: lighten($color: $success-color, $amount: 10%)
  }
}

.error-button {
  background: $error-color;
  &:hover {
    background: darken($color: $error-color, $amount: 10%)
  }
}
