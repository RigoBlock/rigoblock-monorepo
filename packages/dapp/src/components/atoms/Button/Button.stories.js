import '../../_settings/_base.scss'
import { selectV2, text, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import Button, { BUTTON_TYPES } from './Button'
import React from 'react'

const testFunction = () => console.log('button got clicked!')

storiesOf('Atoms/Button', module)
  .addDecorator(withKnobs)
  .add('default', () => (
    <Button
      appearance={selectV2('Button type', BUTTON_TYPES)}
      onClick={testFunction}
    >
      {text('button text', 'example button')}
    </Button>
  ))
