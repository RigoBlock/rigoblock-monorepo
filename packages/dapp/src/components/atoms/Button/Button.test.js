import { shallow } from 'enzyme'
import Button from './Button.jsx'
import toJson from 'enzyme-to-json'

const props = {
  children: 'Test Button',
  onClick: jest.fn()
}

describe('Button component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(Button, props)))
    ).toMatchSnapshot()
  })

  it('calls onClick function when clicked', () => {
    const button = shallow(createComponentWithProps(Button, props))
    button.simulate('click')
    expect(props.onClick).toHaveBeenCalledTimes(1)
  })
})
