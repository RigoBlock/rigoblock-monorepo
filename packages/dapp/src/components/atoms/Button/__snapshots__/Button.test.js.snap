// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<withInk(withTooltip(Button))
  className="b a s e - b u t t o n"
  flat={true}
  inkTransitionEnterTimeout={450}
  inkTransitionLeaveTimeout={300}
  inkTransitionOverlap={150}
  onClick={[MockFunction]}
  primary={true}
  swapTheming={false}
>
  Test Button
</withInk(withTooltip(Button))>
`;
